Jenkinsfile.k8s

```diff
@@ -78,7 +78,14 @@ pipeline {
             export NP=`grep -c ^processor /proc/cpuinfo`
             mkdir -p /home/dealii/build-gcc-fast
             cd /home/dealii/build-gcc-fast
+
+            # We need to add this option to mpirun, otherwise
+            # openmpi complains about running as root. We need
+            # to run as root, because the surrounding Jenkins container
+            # assumes we have root access.
```

Comment:
@#$%^&

Who designed that system? Even in containers, running as root is a poor way of dealing with things...

Reply:
From what I have read in forums discussing containerization, there are quite some people who would disagree with you on that :smirk:. 

Also I want to point out that of course in theory you can use normal users in containers, only our combination (having a jenkins image with a jenkins user id of 10000) that mounts a directory into the dealii image with the dealii user, bundled in containers that run inside kubernetes somewhere produces the problem. Tyler and I spend the better part of two days figuring out a workaround (synchronizing user-ids between the containers, setting access rights on the host filesystem, switching users inside the container), but the only solution we could find was to run as root.

I have no preference about the whole running as root in containers issue, but what I really find a bit frustrating here is that the openmpi project was informed about the complications that their decision to create this command-line option causes, but so far has not decided on a future course of action, because they want to make it intentionally hard to run MPI as root (see also issue 4451 of open-mpi/ompi, I am on purpose not linking to the issue to not get dragged into the discussion :smile:). I can understand their arguments, but it just leads to more conflicts for anyone who wants to run openmpi inside a container image that is designed to run as root. And using an openmpi specific command-line flag that is not necessary in and not even supported by other mpi implementations does not help the problem either.

Reply:
Oh, I hope it was clear that I didn't want to criticize you for this setup. I understood from your previous description that that choice was made for you.

It's exactly the kind of issues you describe why I think running anything as root is a bad idea. Apart from simply making it too easy for anyone to shoot themselves into the foot. But that's definitely not your fault!

Reply:
I left a comment on the open-mpi issue as well.

Reply:
Don't worry, I was aware that you did not criticize our setup but the docker practice of running as root. I just needed to vent a bit of frustration about the fact that the whole jenkins setup got so much more complicated than anticipated. I am curious to see what happens in open-mpi :smile:.

Reply:
This, now, is a little bit funny. I commented on the openmpi issue 4451 with something I didn't think was *completely* serious (though also not entirely stupid), but that's what they implemented and just merged :-)

Reply:
I had to laugh when I saw the discussion :smile:. And I had to read your proposal twice to understand what you meant, so it is definitely complicated enough that nobody does it by accident.

Reply:
Apparently "complicated" is something they value ;-)